examples/plot_metric_examples.py

```diff
@@ -0,0 +1,474 @@
+X_lsml = lsml.fit_transform(X, Y)
+
+plot_tsne(X_lsml, Y)
+
+
+######################################################################
+# Neighborhood Components Analysis
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# NCA is an extremly popular metric-learning algorithm.
+#
+# Neighbourhood components analysis aims at "learning" a distance metric
+# by finding a linear transformation of input data such that the average
+# leave-one-out (LOO) classification performance of a soft-nearest
+# neighbors rule is maximized in the transformed space. The key insight to
+# the algorithm is that a matrix :math:`A` corresponding to the
+# transformation can be found by defining a differentiable objective function
+# for :math:`A`, followed by use of an iterative solver such as
+# `scipy.optimize.fmin_l_bfgs_b`. Like LMNN, this algorithm does not try to
+# cluster points from the same class in a unique cluster, because it
+# enforces conditions at a local neighborhood scale.
+#
+# - See more in the :ref:`User Guide <nca>`
+# - See more in the documentation of the class :py:class:`NCA
+#   <metric_learn.nca.NCA>`
+
+nca = metric_learn.NCA(max_iter=1000)
+X_nca = nca.fit_transform(X, Y)
+
+plot_tsne(X_nca, Y)
+
+######################################################################
+# Local Fischer Discriminant Analysis
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# LFDA is a linear supervised dimensionality reduction method. It is
+# particularly useful when dealing with multimodality, where one ore more
+# classes consist of separate clusters in input space. The core
+# optimization problem of LFDA is solved as a generalized eigenvalue
+# problem. Like LMNN, and NCA, this algorithm does not try to cluster points
+# from the same class in a unique cluster.
+#
+# - See more in the :ref:`User Guide <lfda>`
+# - See more in the documentation of the class :py:class:`LFDA
+#   <metric_learn.lfda.LFDA>`
+
+lfda = metric_learn.LFDA(k=2, num_dims=2)
+X_lfda = lfda.fit_transform(X, Y)
+
+plot_tsne(X_lfda, Y)
+
+
+######################################################################
+# Relative Components Analysis
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# 
+# RCA is another one of the older algorithms. It learns a full rank
+# Mahalanobis distance metric based on a weighted sum of in-class
+# covariance matrices. It applies a global linear transformation to assign
+# large weights to relevant dimensions and low weights to irrelevant
+# dimensions. Those relevant dimensions are estimated using "chunklets",
+# subsets of points that are known to belong to the same class.
+#
+# - See more in the :ref:`User Guide <rca>`
+# - See more in the documentation of the class :py:class:`RCA
+#   <metric_learn.rca.RCA>`
+
+rca = metric_learn.RCA_Supervised(num_chunks=30, chunk_size=2)
+X_rca = rca.fit_transform(X, Y)
+
+plot_tsne(X_rca, Y)
+
+######################################################################
+# Metric Learning for Kernel Regression
```

Comment:
Regression example: Metric Learning for Kernel Regression

Reply:
done